jsonrpc/metrics.go

```diff
@@ -17,6 +17,51 @@ type Metrics struct {
 
 	// Requests duration (seconds)
 	ResponseTime metrics.Histogram
+
+	// Eth metrics
+	ethAPI *stdprometheus.CounterVec
+
+	// Net metrics
+	netAPI *stdprometheus.CounterVec
+
+	// Web3 metrics
+	web3API *stdprometheus.CounterVec
+
+	// TxPool metrics
+	txPoolAPI *stdprometheus.CounterVec
+
+	// Debug metrics
+	debugAPI *stdprometheus.CounterVec
+}
+
+func (m *Metrics) EthAPICounterInc(method string) {
+	if m.ethAPI != nil {
+		m.ethAPI.With(stdprometheus.Labels{"method": "eth_" + method}).Inc()
```

Comment:
why not use constant method names instead of runtime concat?

Reply:
@DarianShawn now use type parameters